src/NuGet.Core/NuGet.Packaging/Signing/Timestamp/Rfc3161TimestampProvider.cs

```diff
@@ -161,7 +161,17 @@ private static void ValidateTimestampCms(SigningSpecifications spec, SignedCms t
 
             if (!CertificateUtility.IsSignatureAlgorithmSupported(signerInfo.Certificate))
             {
-                throw new TimestampException(NuGetLogCode.NU3022, Strings.SignError_TimestampUnsupportedSignatureAlgorithm);
+                var certificateSignatureAlgorithm = signerInfo.Certificate.SignatureAlgorithm.FriendlyName?.ToUpper() ??
```

Comment:
nit: extract this logic to a utility method to reuse it in both places yo do this.